relion/protocols/protocol_subtract.py

```diff
@@ -171,14 +311,20 @@ def createOutputStep(self):
 
         self.reader = convert.createReader(alignType=ALIGN_PROJ)
         tableName = 'particles@' if self.IS_GT30() else ''
+        if self.isRelionInput:
+            doClone = False
+        else:
+            doClone = True
+            self.reader.setPixelSize(3.5)
+
         mdIter = convert.Table.iterRows(tableName + outImgsFn)
-        outImgSet.copyItems(imgSet, doClone=False,
+
+        outImgSet.copyItems(imgSet, doClone=doClone,
```

Comment:
why?

Reply:
I guess the question is why we do not clone when the new version is used and why we do clone when the old version is used. In the new version part of the particle alignment is applied to the particle before the subtraction is made and therefore the "right" alignment for the subtracted particle is the original except the part applied. In the old version the particles were not modified and therefore the alignment information of the input and output particles should be identical (this is handled by clone)

Reply:
Both doClone=True or False the particles are identical, the clone option is more for efficiency reasons.  With doClone=True (the initial implementation and default option) a new Object clone is created during the copyItems iteration.  With doClone=False, the same object is reused during the iteration of copyItems...so in this case, we should keep it False.

Reply:
Let my test this runnin the code with doClone=False.

Reply:
@delarosatrevin you are right doClone=False works. since I see that @azazellochg  is reformating the code I will not modify it but the following change may be included.

--- a/relion/protocols/protocol_subtract.py
+++ b/relion/protocols/protocol_subtract.py
@@ -280,10 +280,6 @@ class ProtRelionSubtract(ProtOperateParticles):
 
         self.reader = convert.createReader(alignType=ALIGN_PROJ)
         tableName = 'particles@' if self.IS_GT30() else ''
-        if self.isRelionInput:
-            doClone = False
-        else:
-            doClone = True
 
         mdIter = convert.Table.iterRows(tableName + outImgsFn)
         if not os.path.exists(outImgsFn):
@@ -291,7 +287,7 @@ class ProtRelionSubtract(ProtOperateParticles):
         else:
             print('----Everything OK-----\n', flush=True)
 
-        outImgSet.copyItems(imgSet, doClone=doClone,
+        outImgSet.copyItems(imgSet, doClone=False,
                             updateItemCallback=self._updateItem,
                             itemDataIterator=mdIter)